zebra-network/src/meta_addr/tests/prop.rs

```diff
@@ -188,4 +210,209 @@ proptest! {
+        let _guard = runtime.enter();
+
+        // Only put valid addresses in the address book.
+        // This means some tests will start with an empty address book.
+        let addrs = if addr.last_known_info_is_valid_for_outbound() {
+            Some(addr)
+        } else {
+            None
+        };
+
+        let address_book = Arc::new(std::sync::Mutex::new(AddressBook::new_with_addrs(&Config::default(), Span::none(), addrs)));
+        let peer_service = service_fn(|_| async { unreachable!("Service should not be called") });
+        let mut candidate_set = CandidateSet::new(address_book.clone(), peer_service);
+
+        runtime.block_on(async move {
+            tokio::time::pause();
+
+            // The earliest time we can have a valid next attempt for this peer
+            let earliest_next_attempt = Instant::now() + LIVE_PEER_DURATION;
+
+            // The number of attempts for this peer in the last LIVE_PEER_DURATION
+            let mut attempt_count: usize = 0;
+
+            for (i, change) in changes.into_iter().enumerate() {
+                while let Some(candidate_addr) = candidate_set.next().await {
+                    assert_eq!(candidate_addr.addr, addr.addr);
+
+                    attempt_count += 1;
+                    assert!(
+                        attempt_count <= 1,
+                        "candidate: {:?}, change: {}, now: {:?}, earliest next attempt: {:?}, attempts: {}, live peer interval limit: {}, test time limit: {:?}, peer change interval: {:?}, original addr was in address book: {}",
+                        candidate_addr,
+                        i,
+                        Instant::now(),
+                        earliest_next_attempt,
+                        attempt_count,
+                        LIVE_PEER_INTERVALS,
+                        overall_test_time,
+                        peer_change_interval,
+                        addr.last_known_info_is_valid_for_outbound(),
+                    );
+                }
+
+                // If `change` is invalid for the current MetaAddr state,
+                // multiple intervals will elapse between actual changes to
+                // the MetaAddr in the AddressBook.
+                address_book.clone().lock().unwrap().update(change);
+
+                tokio::time::advance(peer_change_interval).await;
+                if Instant::now() >= earliest_next_attempt {
+                    attempt_count = 0;
+                }
+            }
+        });
+    }
+
+    /// Make sure that all disconnected [`MetaAddr`]s are retried once, before
+    /// any are retried twice.
+    ///
+    /// This is the simple version of the test, which checks [`MetaAddr`]s by
+    /// themselves. It detects bugs in [`MetaAddr`]s, even if there are
+    /// compensating bugs in the [`CandidateSet`] or [`AddressBook`].
+    //
+    // TODO: write a similar test using the AddressBook and CandidateSet
```

Comment:
This extra test isn't a high priority, because outbound connection fairness itself isn't a high priority.